src/obu.c

```diff
@@ -325,12 +336,59 @@ static avifBool parseSequenceHeader(avifBits * bits, avifSequenceHeader * header
     if (!mono_chrome) {
         avifBitsRead(bits, 1); // separate_uv_delta_q
     }
+
+    return !bits->error;
+}
+
+static avifBool parseAV1SequenceHeader(avifBits * bits, avifSequenceHeader * header)
+{
+    AVIF_CHECK(parseSequenceHeaderProfile(bits, header));
+
+    AVIF_CHECK(parseSequenceHeaderFeatures(bits, header));
+    avifBitsRead(bits, 3); // enable_superres, enable_cdef, enable_restoration
+
+    AVIF_CHECK(parseSequenceHeaderColorConfig(bits, header));
+
     avifBitsRead(bits, 1); // film_grain_params_present
+    return !bits->error;
+}
+
+// See https://gitlab.com/AOMediaCodec/avm/-/blob/main/av1/decoder/decodeframe.c
+static avifBool parseAV2SequenceHeader(avifBits * bits, avifSequenceHeader * header)
+{
+    // See read_sequence_header_obu() in avm.
+    AVIF_CHECK(parseSequenceHeaderProfile(bits, header));
+
+    // See av1_read_sequence_header() in avm.
+    AVIF_CHECK(parseSequenceHeaderFeatures(bits, header));
+    avifBitsRead(bits, 2);       // enable_superres, enable_cdef
+    if (avifBitsRead(bits, 1)) { // enable_restoration
+#if CONFIG_LR_FLEX_SYNTAX
+        avifBitsRead(bits, 2 + CONFIG_PC_WIENER + CONFIG_WIENER_NONSEP); // lr_tools_disable_mask[0]
+        if (avifBitsRead(bits, 1)) {                                     // uv_neq_y
+            avifBitsRead(bits, 2 + CONFIG_WIENER_NONSEP);                // lr_tools_disable_mask[1]
+        }
+#endif
+    }
 
+    // See av1_read_color_config() in avm.
+    AVIF_CHECK(parseSequenceHeaderColorConfig(bits, header));
+    // Ignored fields.
+    //   base_y_dc_delta_q
+    //   base_uv_dc_delta_q
+
+    // See read_sequence_header_obu() in avm.
+    // Ignored field.
+    //   film_grain_params_present
+
+    // See av1_read_sequence_header_beyond_av1() in avm.
+    // Other ignored fields.
     return !bits->error;
 }
 
-avifBool avifSequenceHeaderParse(avifSequenceHeader * header, const avifROData * sample)
+static avifBool avifSequenceHeaderParse(avifSequenceHeader * header,
+                                        const avifROData * sample,
+                                        avifBool (*parse)(avifBits *, avifSequenceHeader *))
```

Comment:
having a function pointer here seems a bit excessive.

given the caller of this function knows whether it's AV1 or AV2, can we just pass that boolean here and add a conditional to call the appropriate function?